web/src/lib/components/utilities-page/duplicates/duplicates-compare-control.svelte

```diff
@@ -0,0 +1,147 @@
+<script lang="ts">
+  import Button from '$lib/components/elements/buttons/button.svelte';
+  import Icon from '$lib/components/elements/icon.svelte';
+  import { locale } from '$lib/stores/preferences.store';
+  import { getAssetThumbnailUrl } from '$lib/utils';
+  import { asByteUnitString } from '$lib/utils/byte-units';
+  import { ThumbnailFormat, type AssetResponseDto, type DuplicateResponseDto, getAllAlbums } from '@immich/sdk';
+  import { mdiCheck, mdiTrashCanOutline } from '@mdi/js';
+  import { onMount } from 'svelte';
+  import { s } from '$lib/utils';
+
+  export let duplicate: DuplicateResponseDto;
+  export let onResolve: (trashIds: string[]) => void;
+
+  let selectedAssetIds = new Set<string>();
+
+  $: trashCount = duplicate.assets.length - selectedAssetIds.size;
+
+  onMount(() => {
+    const suggestedAsset = duplicate.assets.sort((a, b) => {
+      if (!a.exifInfo || !b.exifInfo || !a.exifInfo.fileSizeInByte || !b.exifInfo.fileSizeInByte) {
+        return 0;
+      }
+
+      return b.exifInfo.fileSizeInByte - a.exifInfo.fileSizeInByte;
+    })[0];
+
+    selectedAssetIds.add(suggestedAsset.id);
+    selectedAssetIds = new Set(selectedAssetIds);
+  });
+
+  const onSelectAsset = (asset: AssetResponseDto) => {
+    if (selectedAssetIds.has(asset.id)) {
+      selectedAssetIds.delete(asset.id);
+    } else {
+      selectedAssetIds.add(asset.id);
+    }
+
+    selectedAssetIds = selectedAssetIds;
+  };
+
+  const handleResolve = () => {
+    const trashIds = duplicate.assets.map((asset) => asset.id).filter((id) => !selectedAssetIds.has(id));
+
+    onResolve(trashIds);
+  };
+</script>
+
+<div class="pt-4 rounded-3xl border dark:border-2 border-gray-300 dark:border-gray-700 max-w-[900px] m-auto mb-16">
+  <div class="flex flex-wrap gap-1 place-items-center place-content-center px-4 pt-4">
+    {#each duplicate.assets as asset, index (index)}
+      {@const isSelected = selectedAssetIds.has(asset.id)}
+      {@const isFromExternalLibrary = !!asset.libraryId}
+      {@const assetData = JSON.stringify(asset, null, 2)}
+      {@const hasValidFileSize = asset.exifInfo?.fileSizeInByte != null}
+      {@const hasValidDimensions = asset.exifInfo?.exifImageWidth != null && asset.exifInfo?.exifImageHeight != null}
+      {@const fileSize = hasValidFileSize
+        ? asByteUnitString(Number(asset.exifInfo?.fileSizeInByte), $locale, 4)
+        : 'Invalid Data'}
+      {@const dimensions = hasValidDimensions
+        ? `${asset.exifInfo?.exifImageWidth}x${asset.exifInfo?.exifImageHeight}`
+        : 'Invalid Data'}
```

Comment:
I don't really like having this much logic in the HTML What do you think of a `getFileSize(asset)` function, that returns the byte unit string if it has a valid file size, and undefined (or `'Invalid Data'`) otherwise. The usage would then be `fileSize = getFileSize(asset) || 'Invalid Data'` or `fileSize = getFileSize(asset)` (same thing for dimensions)

Reply:
Sure